src/DonorProfiles/Routes/ProfileRoute.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace Give\DonorProfiles\Routes;
+
+use WP_REST_Request;
+use Give\API\RestRoute;
+use Give\DonorProfiles\Profile as Profile;
+
+/**
+ * @since 2.11.0
+ */
+class ProfileRoute implements RestRoute {
+
+	/** @var string */
+	protected $endpoint = 'donor-profile/profile';
+
+	/**
+	 * @inheritDoc
+	 */
+	public function registerRoute() {
+		register_rest_route(
+			'give-api/v2',
+			$this->endpoint,
+			[
+				[
+					'methods'             => 'POST',
+					'callback'            => [ $this, 'handleRequest' ],
+					'permission_callback' => function() {
+						return is_user_logged_in();
+					},
+				],
+				'args' => [
+					'data' => [
+						'type'              => 'string',
+						'required'          => true,
+						'sanitize_callback' => [ $this, 'sanitizeData' ],
+					],
+					'id'   => [
+						'type'     => 'int',
+						'required' => true,
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Sanitize profile data object
+	 *
+	 * @param string $value JSON encoded string representing profile data
+	 * @param \WP_REST_Request $request
+	 * @param string $param
+	 *
+	 * @return object
+	 *
+	 * @since 2.11.0
+	 */
+	public function sanitizeData( $value, $request, $param ) {
+
+		$sanitizeHelper = '\Give\DonorProfiles\Helpers\SanitizeProfileData';
+		$sanitizedValue = json_decode( $value );
+
+		$attributesMap = [
```

Comment:
Since so many have unique sanitizing strategies., would it be simpler here to procedurally sanitize each key/value? I think the attribute mapping requires a greater mental overhead in this case.

Reply:
This would also reduce the `[ class, method ]` array syntax in favour of an object-oriented syntax, which can be highlighted.

Reply:
This definitely cleared up the function some -- I think my initial hesitancy about a procedural approach related to concerns about setting defaults. But after implementing once, I realize that the default are predictable and I can just set them inside the sanitization functions. Done, ref: https://github.com/impress-org/givewp/pull/5486/commits/66b56f62ca81580ab279ca8df7afa086af7bd1be